shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -407,10 +407,10 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
             err(msg"Method names must start with a capital letter", nme.toLoc)
           if (defs.isDefinedAt(nme.name))
             err(msg"Method '${tn}.${nme.name}' is already defined.", nme.toLoc)
-          (targsMap.keySet & targsMap2.keySet).foreach { s =>
-            warn((s"Method type parameter ${targsMap2(s).nameHint.getOrElse("")}" -> targsMap2(s).prov.loco)
-              :: (s"shadows class type parameter ${td.targsMap(s).nameHint.getOrElse("")}" -> td.targsMap(s).prov.loco)
-              :: Nil map (x => Message.fromStr(x._1) -> x._2))
+          td.tparams.flatMap(tp1 => tparams.find(_.name === tp1.name).map(tp1 -> _)).foreach { case tp1 -> tp2 =>
```

Comment:
Let's not create unnecessary lists in memory! 😛 
Also, better not use a quadratic algorithm when it's so easily avoided...
```suggestion
          val tp1s = td.tparams.toSet
          tparams.foreach { tp2 => tp1s.get(tp2) match ...
```